src/graphql/entries.ts

```diff
@@ -109,51 +111,83 @@ function isAsset(entity: EntryProps | (Entity & CollectionItem)): boolean {
   return 'linkType' in entity.sys && entity.sys.linkType === ASSET_TYPENAME;
 }
 
-function updateReferenceAssetField({
+async function updateReferenceAssetField({
   referenceFromPreviewApp,
   updatedReference,
   entityReferenceMap,
   locale,
 }: UpdateReferenceFieldProps) {
-  const match = entityReferenceMap.get(updatedReference.sys.id) as AssetProps | undefined;
-
-  if (!match) {
-    // if we don't have the asset we send a message back to the entry editor
-    // and it will then send the asset back in the entity reference map
-    // where we can calculate the asset on the next update message.
-    sendMessageToEditor({
-      action: 'ENTITY_NOT_KNOWN',
-      referenceEntityId: updatedReference.sys.id,
-      referenceContentType: ASSET_TYPENAME,
-    });
-    return;
-  }
+  console.log('>> update asset field', { referenceFromPreviewApp, updatedReference });
+  const { match } = await getReference(entityReferenceMap, updatedReference.sys.id, true);
```

Comment:
Being super picky here but maybe makes sense for this to be an object? For example I have no idea what `true` is here at first glance

Reply:
That or creating a separate function for assets

Reply:
yes +1
also don't look to much into the implementation yet, its more about the concept at the current state :D